### PR TITLE
setting ProjectData to be lazy by default

### LINKED ISSUE #835: ProjectData should be lazy by default
It's probably enough to set the lazy flag in the overridden `__init__` and add some tests.

https://github.com/pyiron/pyiron_base/blob/master/pyiron_base/project/data.py

### PR DESCRIPTION
**Pull request by yasser-sulaiman**
In order to improve performance, avoid unnecessary computations, and decrease memory requirements.
I have set the _lazy flag in ProjectData class to True so it is lazy by default.
closes #835

### CODE REVIEW

**Review by liamhuber — COMMENTED** [review]
DataContainer has a "lazy" kwarg, so I don't see a need to set the variable manually; you can just add lazy=True to the super call. Or, maybe better, reproduce the full signature but change the default value for lazy to true and pass lazy=lazy so we still leave the door open to instantiating it non-lazily

**Comment by yasser-sulaiman** [comment]
> DataContainer has a "lazy" kwarg, so I don't see a need to set the variable manually; you can just add lazy=True to the super call. Or, maybe better, reproduce the full signature but change the default value for lazy to true and pass lazy=lazy so we still leave the door open to instantiating it non-lazily

so I have added lazy with default value of True as a parameter to the ProjectData's constructor, is that what you meant?

File: pyiron_base/project/data.py
Comment (pmrv) [inline pyiron_base/project/data.py]:
```suggestion
        super().__init__(*args, lazy=lazy, **kwargs)
```

This is probably what @liamhuber meant.

Comment (yasser-sulaiman) [reply pyiron_base/project/data.py]: ok then, thanks

Comment (yasser-sulaiman) [reply pyiron_base/project/data.py]: so what happens now?
will the issue [#835](https://github.com/pyiron/pyiron_base/issues/835) be closed?

**Review by liamhuber — APPROVED** [review]

**Comment by liamhuber** [comment]
> so what happens now?
> will the issue https://github.com/pyiron/pyiron_base/issues/835 be closed?

I flagged it to run the integration tests, I don't anticipate any problems but for the sake of completeness we'll let them run. Once they all pass I'll merge this and you can indeed close #835 😄

**Comment by pmrv** [comment]
Sorry, I lost track of this.  Since the tests pass now, I think we can merge it?

**Comment by yasser-sulaiman** [comment]
I am waiting for it to be merged actually @pmrv :)

**Comment by pmrv** [comment]
> I am waiting for it to be merged actually @pmrv :)

Sorry to keep you waiting. Some things just drop off the radar for a bit sometimes.  Feel free to ping me next time. :)

**Pull request merged**
